cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -442,53 +464,55 @@ def test_provision_with_boot_volume(request, instance_args, provider, soft_asser
+                        :delete_on_termination => false
+                    }}]
+                }}
+            )
+        '''.format(volume))
+
+    @request.addfinalizer
+    def _finish_method():
         with update(method):
-            method.script = dedent('''\
-                $evm.root["miq_provision"].set_option(
-                    :clone_options, {{
-                        :image_ref => nil,
-                        :block_device_mapping_v2 => [{{
-                            :boot_index => 0,
-                            :uuid => "{}",
-                            :device_name => "vda",
-                            :source_type => "volume",
-                            :destination_type => "volume",
-                            :volume_size => 1,
-                            :delete_on_termination => false
-                        }}]
-                    }}
-                )
-            '''.format(volume))
-
-        @request.addfinalizer
-        def _finish_method():
-            with update(method):
-                method.script = """prov = $evm.root["miq_provision"]"""
-
-        instance = appliance.collections.cloud_instances.create(vm_name,
-                                                                provider,
-                                                                form_values=inst_args)
-
-        request_description = 'Provision from [{}] to [{}]'.format(image,
-                                                                   instance.name)
-        provision_request = appliance.collections.requests.instantiate(request_description)
-        try:
-            provision_request.wait_for_request(method='ui')
-        except Exception as e:
-            logger.info(
-                "Provision failed {}: {}".format(e, provision_request.request_state))
-            raise
-        msg = "Provisioning failed with the message {}".format(
-            provision_request.row.last_message.text)
-        assert provision_request.is_succeeded(method='ui'), msg
-        soft_assert(instance.name in provider.mgmt.volume_attachments(volume))
-        soft_assert(provider.mgmt.volume_attachments(volume)[instance.name] == "/dev/vda")
+            method.script = former_method_script
+
+    instance = appliance.collections.cloud_instances.create(vm_name,
+                                                            provider,
+                                                            form_values=inst_args)
+
+    @request.addfinalizer
+    def delete_vm_and_wait_for_gone():
         instance.mgmt.delete()  # To make it possible to delete the volume
         wait_for(lambda: not instance.exists_on_provider, num_sec=180, delay=5)
 
+    request_description = 'Provision from [{}] to [{}]'.format(image, instance.name)
+    provision_request = appliance.collections.requests.instantiate(request_description)
+    provision_request.wait_for_request(method='ui')
+
+    msg = "Provisioning failed with the message {}".format(
```

Comment:
At the risk of being snarky, above you disagreed with @john-dupuy suggestion to use constants for the ruby scripts on the basis of not polluting the namespace with named variables that are used once. Here you've named `request_description` and `msg` when they're only used once, immediately on the next line. Why not put these strings directly into their relevant function calls?

Reply:
1) It seems to me that both cases are existing code that I was only moving around or shifting left/right.
2) These variables have limited scope and they are used immediately after their definition. Therefore one doesn't need to scroll too mulch to see their definition.

So what I would like to do rather then placing them on top of the file is to place them near they one-shot-only use. They don't really need to be defined in modscope AFAIK